reactive/kubernetes_worker.py

```diff
@@ -751,6 +751,13 @@ def configure_kubelet(dns, ingress_ip):
     if get_version('kubelet') >= (1, 11):
         kubelet_opts['dynamic-config-dir'] = '/root/cdk/kubelet/dynamic-config'
 
+    if is_state('kube-control.registry_location.available'):
+        kube_control = endpoint_from_flag(
+            'kube-control.registry_location.available')
+        registry_location = kube_control.get_registry_location().rstrip('/')
+        kubelet_opts['pod-infra-container-image'] = \
```

Comment:
can registry_location be an empty string? I know you can manually set the config to empty. Should we check for that here?

Reply:
@hyperbolic2346 i don't think it can be. We only get the `.available` flag if it's truthy:

https://github.com/juju-solutions/interface-kube-control/blob/master/requires.py#L51

That said, someone could config a truthy registry, and then reconfigure it to be empty -- at that point, maybe we need to reconfigure kubelet to make sure `kubelet_opts['pod-infra-container-image']` is not set?  So like an `else` block here?

Reply:
Fwiw, I took another look here, and we wipe out `kubelet_opts` during each entry into `configure_kubelet`. We won't need an `else` here because opts will only include this config if the flag is present, and the presence of the flag means there's a truthy registry.